contracts/Account.sol

```diff
@@ -398,17 +447,28 @@ contract Account is UUPSOwnableUpgradeable, UsingRegistryUpgradeable, Managed, I
         }
 
         // If there is no CELO to lock up and vote with, return.
-        if (unlockedCeloForGroup == 0) {
+        if (celoToVoteForGroup == 0) {
             return;
         }
 
+        uint256 accountLockedUnvotingGold = getLockedGold().getAccountNonvotingLockedGold(
+            address(this)
+        );
+
+        // There might be some locked unvoting (revoked) gold from previous transfers
+        uint256 toLock = accountLockedUnvotingGold >= celoToVoteForGroup
+            ? 0
+            : celoToVoteForGroup - accountLockedUnvotingGold;
+
         // Lock up the unlockedCeloForGroup in LockedGold, which increments the
         // non-voting LockedGold balance for this contract.
-        getLockedGold().lock{value: unlockedCeloForGroup}();
+        if (toLock > 0) {
+            getLockedGold().lock{value: toLock}();
+        }
 
         // Vote for group using the newly locked CELO, reverting if it fails.
-        if (!election.vote(group, unlockedCeloForGroup, voteLesser, voteGreater)) {
-            revert VoteFailed(group, unlockedCeloForGroup);
+        if (!election.vote(group, celoToVoteForGroup, voteLesser, voteGreater)) {
```

Comment:
FYI, currently `Election.vote` never returns `false`. So if we want to detect e.g. not having enough votes, we should add sanity checks before this call.